common4j/src/main/com/microsoft/identity/common/java/opentelemetry/CryptoFactoryTelemetryHelper.java

```diff
@@ -22,59 +22,67 @@
+import static com.microsoft.identity.common.java.opentelemetry.AttributeName.parent_span_name;
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
 
 import com.microsoft.identity.common.java.crypto.ICryptoFactory;
 import com.microsoft.identity.common.java.exception.ClientException;
-import com.microsoft.identity.common.java.util.StringUtil;
+import com.microsoft.identity.common.java.exception.IErrorInformation;
+import com.microsoft.identity.common.java.util.ThrowableUtil;
 
-import io.opentelemetry.api.trace.Span;
-import io.opentelemetry.api.trace.StatusCode;
-import io.opentelemetry.context.Scope;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
 import lombok.NonNull;
 
 public class CryptoFactoryTelemetryHelper {
 
+    private static final LongCounter sFailedCryptoOperationCount = OTelUtility.createLongCounter(
+            "failed_crypto_operation_count",
+            "Number of failed crypto operations"
+    );
+
     /**
      * A helper class that consolidate all the telemetry emitting work
      * for crypto operation in one place.
      *
      * @param cryptoOperation name of the crypto operation.
-     * @param algorithmName name of the algorithm.
-     * @param cryptoFactory an {@link ICryptoFactory} object.
+     * @param algorithmName   name of the algorithm.
+     * @param cryptoFactory   an {@link ICryptoFactory} object.
      * @param cryptoOperation a callback that wraps around the crypto operation to be performed.
      * @return result of the crypto operation.
-     * */
+     */
     public static <T> T performCryptoOperationAndUploadTelemetry(@NonNull final CryptoObjectName operationName,
                                                                  @NonNull final String algorithmName,
                                                                  @NonNull final ICryptoFactory cryptoFactory,
                                                                  @NonNull final ICryptoOperation<T> cryptoOperation)
             throws ClientException {
-        final Span span = OTelUtility.createSpan(SpanName.CryptoFactoryEvent.name());
-        try (final Scope scope = span.makeCurrent()) {
-            span.setAttribute(crypto_controller.name(), cryptoFactory.getTelemetryClassName().name());
-            span.setAttribute(crypto_operation.name(),
-                    getCryptoOperationEventName(operationName, algorithmName));
-            span.setStatus(StatusCode.OK);
+        try {
             return cryptoOperation.perform();
         } catch (final Exception e) {
-            span.setStatus(StatusCode.ERROR);
-            span.recordException(e);
-            span.setAttribute(crypto_exception_stack_trace.name(),
-                    StringUtil.getStackTraceAsString(e));
+            final String parentSpanName = OTelUtility.getCurrentSpanName();
+            final Attributes attributes = Attributes.of(
```

Comment:
If there be a need to calculate the % number for threshold, we would need the total cases as well, my guess would be we are also storing that data, and not just the failed cases for cryptos?